pkg/controller/utils/utils.go

```diff
@@ -986,3 +987,49 @@ func RemoveInstallationFinalizer(i *operatorv1.Installation, finalizer string) {
 		i.SetFinalizers(stringsutil.RemoveStringInSlice(finalizer, i.GetFinalizers()))
 	}
 }
+
+// maintainInstallationFinalizer manages a controller's finalizer on the Installation resource.
+// We add a finalizer to the Installation when the mainResource has been installed, and only remove that finalizer when
+// the resource has been deleted and its secondary resources have stopped running. This allows for a graceful cleanup of any resources
+// prior to the CNI plugin being removed.
+func MaintainInstallationFinalizer(ctx context.Context, c client.Client, mainResource client.Object,
+	secondaryResource client.Object, finalizer string) error {
```

Comment:
```suggestion
func MaintainInstallationFinalizer(
  ctx context.Context, 
  c client.Client, 
  mainResource client.Object,
  secondaryResource client.Object, 
  finalizer string,
) error {
```

If splitting across lines, please put each argument on its own line for legibility.

Reply:
I would probably suggest for allowing an arbitrary number of secondary resources, though. Some controllers do manage multiple deployments, for example. 

```go
func MaintainInstallationFinalizer(
  ctx context.Context, 
  c client.Client, 
  mainResource client.Object,
  finalizer string,
  objs ...client.Object, 
) error {